packages/vue/src/TolgeeProvider.ts

```diff
@@ -18,28 +20,69 @@ export const TolgeeProvider = defineComponent({
     fallback: {
       type: [Object, String] as PropType<JSX.Element | string>,
     },
+    staticData: {
+      type: Object as PropType<TolgeeStaticData>,
+      required: false,
+    },
+    language: {
+      type: String as PropType<string>,
+      required: false,
+    },
   },
 
   setup(props) {
-    const tolgee: TolgeeInstance | undefined =
-      props.tolgee || getCurrentInstance().proxy.$tolgee;
+    const tolgeeContext: Ref<TolgeeVueContext> = inject('tolgeeContext');
+
+    // for backward compatibility
+    if (props.tolgee) {
+      tolgeeContext.value.tolgee = props.tolgee;
+    }
+
+    const tolgee: ComputedRef<TolgeeInstance> = computed(
+      () => tolgeeContext.value.tolgee
+    );
 
-    if (!tolgee) {
+    if (!tolgee.value) {
       throw new Error('Tolgee instance not provided');
     }
 
-    provide('tolgeeContext', { tolgee } as TolgeeVueContext);
+    if (tolgeeContext.value.isInitialRender) {
+      tolgee.value.setEmitterActive(false);
+      tolgee.value.addStaticData(props.staticData);
+      tolgee.value.changeLanguage(props.language);
```

Comment:
I think here, it needs to throw an error if user doesn't specify the language. Because if we don't know the language, there might be mismatch on server vs client.

Reply:
Actually, we might make it mandatory to specify both `language` and `staticData` when `isInitialRender` is true.

Reply:
Good point. Added